core/src/DynamicSchema.ts

```diff
@@ -149,17 +149,19 @@ async function createDynamicSchema(
 
   const { schemaName, schemaAlias } = props;
   const newSchema = new MetaSchema(context, schemaName, schemaAlias, version.readVersion, version.writeVersion, version.minorVersion);
-
-  const loader = new IModelSchemaLoader(db);
-  const bisSchema = loader.getSchema("BisCore");
+  const bisSchema = await tryGetSchema(db, "BisCore");
   await context.addSchema(newSchema);
-  await context.addSchema(bisSchema);
-  await (newSchema as MutableSchema).addReference(bisSchema); // TODO remove this hack later
+  if (bisSchema != undefined) {
+    await context.addSchema(bisSchema);
+    await (newSchema as MutableSchema).addReference(bisSchema); // TODO remove this hack later
+    }
 
```

Comment:
And if it is `undefined`, this seems worth at least a log, or perhaps even an exception? Can we reasonably continue without BisCore?

Reply:
Is this really only worth a `logInfo`? Like... if you simulate the failure, does the connector work at all without BisCore?

Reply:
Please let me have another look.

Reply:
@jffmarker Thanks for pointing this out and having me revisit.  I changed both locations to throw error.  Resulting unit tests when schema names are altered in debugger to ensure they aren't found are ...

 1) Unit Tests
       synchronize StandaloneDb:
     Error: Schema Name, 'BisCore' not found in iModel.
      at createDynamicSchema (lib\DynamicSchema.js:7:66)
      at async syncDynamicSchema (lib\DynamicSchema.js:5:1289)
      at async JSONConnector._updateDynamicSchema (lib\PConnector.js:9:6019)
      at async JSONConnector.runJobUnsafe (lib\PConnector.js:9:1829)
      at async Context.<anonymous> (lib\test\unit\PConnector.test.js:100:17)

and for the second call to TryGetSchema ...

  1) Unit Tests
       synchronize StandaloneDb:
     Error: Schema Name, 'AecUnits' not found in iModel.
      at createDynamicSchema (lib\DynamicSchema.js:7:526)
      at async syncDynamicSchema (lib\DynamicSchema.js:5:1289)
      at async JSONConnector._updateDynamicSchema (lib\PConnector.js:9:6019)
      at async JSONConnector.runJobUnsafe (lib\PConnector.js:9:1829)
      at async Context.<anonymous> (lib\test\unit\PConnector.test.js:100:17)